WorkloadManagementSystem/DB/PilotAgentsDB.py

```diff
@@ -632,6 +634,96 @@ def getPilotSummary(self, startdate='', enddate=''):
+    based on the same algorithm as in the Web version, basically takes into account Done and
+    Aborted pilots only from the last day. The selection is done entirely in SQL.
+
+    :param selectDict: A dictionary to pass additional conditions to select statements, i.e.
+                       it allows to define start time for Done and Aborted Pilots.
+    :param columnList  a list of column to consider when grouping to calculate efficiencies.
+                       e.g. ['GridSite', 'DestinationSite'] is used to calculate efficiences
+                       for sites and  CEs. If we want to add an OwnerGroup it would be:
+                       ['GridSite', 'DestinationSite', 'OwnerGroup'].
+    :return: a dict containing the ParameterNames and Records lists.
+    """
+
+    table = PivotedPilotSummaryTable(columnList)
+    sqlQuery = table.buildSQL()
+
+    self.logger.info("SQL query : ")
+    self.logger.info("\n" + sqlQuery)
+
+    res = self._query(sqlQuery)
+    if not res['OK']:
+      return res
+
+    self.logger.info(res)
+    # TODO add site or CE status, while looping
+    rows = []
+    columns = table.getColumnList()
+    try:
+      groupIndex = columns.index('OwnerGroup')
+      # should probably change a column name to VO here as well to avoid confusion
+    except ValueError:
+      groupIndex = None
+    result = {'ParameterNames': columns}
+    multiple = False
+    # If not grouped by CE:
+    if 'CE' not in columns:
+      multiple = True
+
+    for row in res['Value']:
+      lrow = list(row)
+      if groupIndex:
+        lrow[groupIndex] = getVOForGroup(row[groupIndex])
+      if multiple:
+        lrow.append('Multiple')
+      for index, value in enumerate(row):
+        if isinstance(value, decimal.Decimal):
+          lrow[index] = float(value)
+      # get the value of the Total column
+      if 'Total' in columnList:
+        total = lrow[columnList.index('Total')]
+      else:
+        total = 0
+      if 'PilotJobEff' in columnList:
+        eff = lrow[columnList.index('PilotJobEff')]
+      else:
+        eff = 0.
+      lrow.append(self._getElementStatus(total, eff))
+      rows.append(tuple(lrow))
+# If not grouped by CE and more then 1 CE in the result:
+    if multiple:
+      columns.append('CE')  # 'DestinationSite' re-mapped to 'CE' already
+    columns.append('Status')
+    result['Records'] = rows
+
+    return S_OK(result)
+
+  def _getElementStatus(self, total, eff):
+    """
+    Assign status to a site or resource based on pilot efficiency.
+    :param total: number of pilots to assign the status, otherwise 'Idle'
+    :param eff:  efficiency in %
+
+    :return: status string
+    """
+
+    # Evaluate the quality status of the Site/CE
+    if total > 10:
+      if eff < 25.:
+        return 'Bad'
+      elif eff < 60.:
+        return 'Poor'
+      elif eff < 85.:
+        return 'Fair'
+      else:
+        return 'Good'
```

Comment:
Are these "adjectives" used somewhere?

Reply:
Not my invention. This is still present in the current integration branch. These values are listed in a _status_ column of the PilotCache table. For information purposes, I believe. Only numeric values are used to determine the quality of the CE.